vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/codec/json/JsonObjectCodec.java

```diff
@@ -106,6 +149,14 @@ protected Object readObjectId(BsonReader reader, DecoderContext ctx) {
 
   @Override
   protected Object readDateTime(BsonReader reader, DecoderContext ctx) {
```

Comment:
+1. I think it makes sense to merge the PR as it is now, and then support other formats in an other PR. 

From my point of view, it is relevant to support all possible input formats (`JsonObjectCodec#writeDateTime`). But for output (`JsonObjectCodec#readDateTime`) we need to choose one, and the ISO format seems to be the more explicit.